osquery/tables/system/darwin/user_groups.mm

```diff
@@ -0,0 +1,107 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <vector>
+#include <string>
+
+#include <grp.h>
+#include <pwd.h>
+
+#import <OpenDirectory/OpenDirectory.h>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/tables/system/user_groups.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genUserGroups(QueryContext &context) {
+  @autoreleasepool {
```

Comment:
I wonder if we can abstract the `ODQuery` and result array iteration into a utility method in `user_groups`? It might need two implementation files, one ObjC++/arc the other C++.